File: app/src/main/java/io/mdevlab/unconnectify/connectivitymodels/Bluetooth.java
package io.mdevlab.unconnectify.connectivitymodels;

import android.bluetooth.BluetoothAdapter;

/**
 * This class handles connecting to and disconnecting from Bluetooth
 * <p>
 * Bluetooth implements the singleton design pattern since only one instance of this class is needed
 * to manage enabling/disabling Bluetooth
 * <p>
 * Created by mdevlab on 2/10/17.
 */

public class Bluetooth extends Connectivity {

    private static Bluetooth instance = null;
    private BluetoothAdapter mBluetoothAdapter = null;

    private Bluetooth() {
        mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
    }

    public static Bluetooth getInstance() {
        if (instance == null)
            instance = new Bluetooth();
        return instance;
    }

    public void enable() {
        if (!mBluetoothAdapter.isEnabled())
            mBluetoothAdapter.enable();
    }

    public void disable() {
        if (mBluetoothAdapter.isEnabled())
            mBluetoothAdapter.disable();
    }
}

File: app/src/main/java/io/mdevlab/unconnectify/connectivitymodels/Bluetooth.kt
package io.mdevlab.unconnectify.connectivitymodels

import android.bluetooth.BluetoothAdapter


class Bluetooth private constructor() : Connectivity() {

    private val bluetoothAdapter: BluetoothAdapter by lazy {
        BluetoothAdapter.getDefaultAdapter()
    }

    override fun enable() {
        if (!bluetoothAdapter.isEnabled) {
            bluetoothAdapter.enable()
        }
    }

    override fun disable() {
        if (bluetoothAdapter.isEnabled) {
            bluetoothAdapter.disable()
        }
    }

    companion object {
        private var instance: Bluetooth? = null
        fun getInstance(): Bluetooth {
            if (instance == null) {
                instance = Bluetooth()
            }
            return instance!!
        }
    }
}
